File: VB/WebApplication1/CustomControls/CustomAppointmentSaveCallbackCommand .vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.UI
Imports DevExpress.Web
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.Web.ASPxScheduler.Internal
Imports DevExpress.XtraScheduler.Native

Namespace WebApplication1.CustomControls
	Public Class CustomAppointmentSaveCallbackCommand
		Inherits AppointmentFormSaveCallbackCommand
		Public Sub New(ByVal control As ASPxScheduler)
			MyBase.New(control)
		End Sub

		Protected Friend Shadows ReadOnly Property Controller() As CustomAppointmentFormController
			Get
				Return CType(MyBase.Controller, CustomAppointmentFormController)
			End Get
		End Property

		Protected Overrides Function FindControlByID(ByVal id As String) As System.Web.UI.Control
			Return FindTemplateControl(TemplateContainer, id)
		End Function

		Private Function FindTemplateControl(ByVal RootControl As System.Web.UI.Control, ByVal id As String) As System.Web.UI.Control
			Dim foundedControl As System.Web.UI.Control = RootControl.FindControl(id)
			If foundedControl Is Nothing Then
				For Each item As System.Web.UI.Control In RootControl.Controls
					foundedControl = FindTemplateControl(item, id)
					If foundedControl IsNot Nothing Then
						Exit For
					End If
				Next item
			End If
			Return foundedControl
		End Function

		Protected Overrides Sub AssignControllerValues()
			MyBase.AssignControllerValues()

			Dim tbSubject As ASPxTextBox = CType(FindControlByID("ASPxTextBoxSubject"), ASPxTextBox)
			Dim deStart As ASPxDateEdit = CType(FindControlByID("ASPxDateEditStart"), ASPxDateEdit)
			Dim deEnd As ASPxDateEdit = CType(FindControlByID("ASPxDateEditEnd"), ASPxDateEdit)
			Dim tbDescription As ASPxMemo = CType(FindControlByID("ASPxMemoDescription"), ASPxMemo)
			Dim tbCustomInfo As ASPxTextBox = CType(FindControlByID("ASPxTextBoxCustomInfo"), ASPxTextBox)

			Controller.Subject = tbSubject.Text
			Controller.Start = Convert.ToDateTime(deStart.Value)
			Controller.End = Convert.ToDateTime(deEnd.Value)
			Controller.Description = tbDescription.Text
			Controller.CustomInfoField = tbCustomInfo.Text

			Controller.CustomAttachment = CType(HttpContext.Current.Session("AppointmentAttachment"), Byte())
			Controller.CustomAttachmentFileName = If(HttpContext.Current.Session("AppointmentAttachmentFileName") IsNot Nothing, HttpContext.Current.Session("AppointmentAttachmentFileName").ToString(), "")

            If Controller.ResourceId Is DevExpress.XtraScheduler.ResourceEmpty.Id Then
                Controller.ResourceId = Control.SelectedResource.Id
            End If
		End Sub

		Protected Overrides Function CreateAppointmentFormController(ByVal apt As DevExpress.XtraScheduler.Appointment) As AppointmentFormController
			Return New CustomAppointmentFormController(Control, apt)
		End Function
	End Class
End Namespace

File: CS/WebApplication1/CustomControls/CustomAppointmentSaveCallbackCommand .cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using DevExpress.Web;
using DevExpress.Web.ASPxScheduler;
using DevExpress.Web.ASPxScheduler.Internal;
using DevExpress.XtraScheduler.Native;

namespace WebApplication1.CustomControls {
    public class CustomAppointmentSaveCallbackCommand : AppointmentFormSaveCallbackCommand {
        public CustomAppointmentSaveCallbackCommand(ASPxScheduler control) : base(control) { }

        protected internal new CustomAppointmentFormController Controller {
            get { return (CustomAppointmentFormController)base.Controller; }
        }

        protected override System.Web.UI.Control FindControlByID(string id) {
            return FindTemplateControl(TemplateContainer, id);
        }

        System.Web.UI.Control FindTemplateControl(System.Web.UI.Control RootControl, string id) {
            System.Web.UI.Control foundedControl = RootControl.FindControl(id);
            if(foundedControl == null) {
                foreach(System.Web.UI.Control item in RootControl.Controls) {
                    foundedControl = FindTemplateControl(item, id);
                    if(foundedControl != null) break;
                }
            }
            return foundedControl;
        }

        protected override void AssignControllerValues() {
            base.AssignControllerValues();

            ASPxTextBox tbSubject = (ASPxTextBox)FindControlByID("ASPxTextBoxSubject");
            ASPxDateEdit deStart = (ASPxDateEdit)FindControlByID("ASPxDateEditStart");
            ASPxDateEdit deEnd = (ASPxDateEdit)FindControlByID("ASPxDateEditEnd");
            ASPxMemo tbDescription = (ASPxMemo)FindControlByID("ASPxMemoDescription");
            ASPxTextBox tbCustomInfo = (ASPxTextBox)FindControlByID("ASPxTextBoxCustomInfo");

            Controller.Subject = tbSubject.Text;
            Controller.Start = Convert.ToDateTime(deStart.Value);
            Controller.End = Convert.ToDateTime(deEnd.Value);
            Controller.Description = tbDescription.Text;
            Controller.CustomInfoField = tbCustomInfo.Text;

            Controller.CustomAttachment = (byte[])HttpContext.Current.Session["AppointmentAttachment"];
            Controller.CustomAttachmentFileName = HttpContext.Current.Session["AppointmentAttachmentFileName"] != null ? HttpContext.Current.Session["AppointmentAttachmentFileName"].ToString() : "";

            if(Controller.ResourceId == DevExpress.XtraScheduler.ResourceEmpty.Id) {
                Controller.ResourceId = Control.SelectedResource.Id;
            }
        }

        protected override AppointmentFormController CreateAppointmentFormController(DevExpress.XtraScheduler.Appointment apt) {
            return new CustomAppointmentFormController(Control, apt);
        }
    }
}